src/Illuminate/Contracts/Redis/Connector.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Illuminate\Contracts\Redis;
+
+interface Connector
+{
+    /**
+     * Create a new clustered redis connection.
+     *
+     * @param  array  $config
+     * @param  array  $options
+     * @return \Illuminate\Redis\Connections\Connection
```

Comment:
I've realized that `\Illuminate\Contracts\Redis\Connection` is not sufficient.

RedisManager has a method type hinted on `\Illuminate\Redis\Connections\Connection`.

It also appears that `\Illuminate\Contracts\Redis\Connection` does not have all of the methods defined in `\Illuminate\Redis\Connections\Connection`.

So extensions will need to subclass `\Illuminate\Redis\Connections\Connection`. That does not seems to be a problem as there are no private methods or properties.

The ideal solution would be to change the interface but that would break backwards compatibility.

We can also add a new interface inheriting from this one `\Illuminate\Contracts\Redis\Connection`. But having two interface for the same thing does not sounds ideal.